core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/ExternalDatabaseService.java

```diff
@@ -144,8 +158,12 @@ public void upgradeDatabase(ClusterView cluster, UpgradeTargetMajorVersion targe
                 try {
                     UpgradeDatabaseServerV4Request request = new UpgradeDatabaseServerV4Request();
                     request.setUpgradeTargetMajorVersion(targetMajorVersion);
-                    redbeamsClient.upgradeByCrn(databaseCrn, request);
-                    waitAndGetDatabase(cluster, databaseCrn, DatabaseOperation.UPGRADE, false);
+                    UpgradeDatabaseServerV4Response response = redbeamsClient.upgradeByCrn(databaseCrn, request);
+                    if (null == response.getFlowIdentifier()) {
+                        LOGGER.info(response.getReason());
```

Comment:
A more elaborate log message would be helpful here, indicating that no flowID returned means that the upgrade is already finished on the redbeams side.

Reply:
If the flowIdentifier is null, then the reason coming from redbeams contains just that.